zerver/lib/upload.py

```diff
@@ -269,16 +272,10 @@ def realm_avatar_and_logo_path(self, realm: Realm) -> str:
 
 ### S3
 
-def get_bucket(conn: S3Connection, bucket_name: str) -> Bucket:
-    # Calling get_bucket() with validate=True can apparently lead
-    # to expensive S3 bills:
-    #    https://www.appneta.com/blog/s3-list-get-bucket-default/
-    # The benefits of validation aren't completely clear to us, and
-    # we want to save on our bills, so we set the validate flag to False.
-    # (We think setting validate to True would cause us to fail faster
-    #  in situations where buckets don't exist, but that shouldn't be
-    #  an issue for us.)
-    bucket = conn.get_bucket(bucket_name, validate=False)
+def get_bucket(session: Session, bucket_name: str) -> ServiceResource:
+    # See https://github.com/python/typeshed/issues/2706
+    # for why this return type is a `ServiceResource`.
+    bucket = session.resource('s3', verify=False).Bucket(bucket_name)
```

Comment:
Yeah, we should remove `verify` options so we don’t disable SSL verification. Line commenting to make sure we don’t forget.

The unrelated Boto 2 `validate=False` option appears to be the default behavior of Boto 3.

Reply:
Cool, thanks Anders. And my apology for the oversight.